### PR TITLE
App status report

### PR DESCRIPTION
**Pull request by emord**
@benrudolph probably still slow, but at least it should work with sql domains

cc: @esoergel

### CODE REVIEW

File: corehq/apps/reports/util.py [hunk 1]
```diff
@@ -11,11 +11,7 @@
 from corehq.apps.users.permissions import get_extra_permissions
 
 from couchexport.util import SerializableFunction
-from couchforms.analytics import (
-    get_all_user_ids_submitted,
-    get_first_form_submission_received,
-    get_username_in_last_form_user_id_submitted,
-)
+from couchforms.analytics import get_first_form_submission_received
```
Comment (emord) [inline corehq/apps/reports/util.py]: This last one is used in the datespan filter, but it's fixed when someone uses new exports feature flag, so not touching right now

File: corehq/apps/reports/standard/deployments.py [hunk 2]
```diff
@@ -96,6 +99,22 @@ def headers(self):
         headers.custom_sort = [[1, 'desc']]
         return headers
 
+
+    @property
```
Comment (dimagimon) [inline corehq/apps/reports/standard/deployments.py]: E303 too many blank lines (2)

File: corehq/apps/reports/standard/deployments.py [hunk 3]
```diff
@@ -151,7 +170,7 @@ def rows(self):
                 last_sync = last_sync_log.date
 
             rows.append(
-                [user.username_in_report, _fmt_date(last_seen), _fmt_date(last_sync), app_name or "---"]
+                [user.username_in_report, _fmt_date(last_seen)['html'], _fmt_date(last_sync)['html'], app_name or "---"]
```
Comment (dimagimon) [inline corehq/apps/reports/standard/deployments.py]: E501 line too long (120 > 115 characters)

**Comment by emord** [comment]
updated with hopeful speed up for http://manage.dimagi.com/default.asp?227230

**Comment by benrudolph** [comment]
woot nice! i've been testing all the new ES accessors here: https://github.com/dimagi/commcare-hq/blob/master/corehq/apps/reports/tests/test_esaccessors.py. would be great to add a couple

File: corehq/apps/reports/analytics/esaccessors.py [hunk 4]
```diff
@@ -473,3 +473,24 @@ def _domain_stats(domain_name):
         domain: _domain_stats(domain)
         for domain in domains
     }
+
+
+def get_all_user_ids_submitted(domain, app_ids=None):
+    query = (
+        FormES()
+        .domain('demo')
```
Comment (benrudolph) [inline corehq/apps/reports/analytics/esaccessors.py]: `domain`

File: corehq/apps/reports/util.py (hunk 1 above)
Comment (benrudolph) [inline corehq/apps/reports/util.py]: 👍

File: corehq/ex-submodules/couchforms/tests/test_analytics.py [hunk 5]
```diff
@@ -6,13 +6,14 @@
...
-    get_all_xmlns_app_id_pairs_submitted_to_in_domain, \
-    get_form_analytics_metadata, get_exports_by_form
+from couchforms.analytics import (
+    domain_has_submission_in_last_30_days,
+    get_number_of_forms_per_domain, get_number_of_forms_in_domain,
+    get_first_form_submission_received, get_last_form_submission_received,
+    app_has_been_submitted_to_in_last_30_days, update_analytics_indexes,
+    get_all_xmlns_app_id_pairs_submitted_to_in_domain,
+    get_form_analytics_metadata, get_exports_by_form,
+)
```
Comment (benrudolph) [inline corehq/ex-submodules/couchforms/tests/test_analytics.py]: ❤️

File: corehq/apps/reports/analytics/esaccessors.py [hunk 6]
```diff
@@ -473,3 +473,24 @@ def _domain_stats(domain_name):
...
+
+    if app_ids:
+        query = query.app(app_ids)
+
+    return query.run().aggregations.user_id.buckets_dict.keys()
+
+
+def get_username_in_last_form_user_id_submitted(domain, user_id):
+    last_sub = get_last_form_submissions_by_user(domain, [user_id])[user_id][0]
+    return last_sub['form']['meta']['username']
```
Comment (esoergel) [inline corehq/apps/reports/analytics/esaccessors.py]: I like this abstraction!

File: corehq/apps/reports/tests/test_export_api.py [hunk 7]
```diff
@@ -72,19 +80,56 @@ def setUp(self):
...
+            f = get_form()
+        url = get_submit_url(domain)
+        self._send_form_to_es(
+            domain=domain,
+            xmlns="http://www.commcarehq.org/export/test",
+        )
+        return spoof_submission(url, f)
+
+
+    def _pillow_process_form(self, form_pair):
```
Comment (dimagimon) [inline corehq/apps/reports/tests/test_export_api.py]: E303 too many blank lines (2)

File: corehq/apps/reports/tests/test_util.py [hunk 8]
```diff
@@ -1,13 +1,21 @@
+from django.test import TestCase
+from elasticsearch.exceptions import ConnectionError
+from mock import Mock, patch
```
Comment (dimagimon) [inline corehq/apps/reports/tests/test_util.py]: F401 'patch' imported but unused

File: corehq/apps/reports/tests/test_export_api.py [hunk 9]
```diff
@@ -1,19 +1,32 @@
...
-from corehq.apps.accounting.tests import BaseAccountingTest
 from corehq.apps.accounting.models import SoftwarePlanEdition
+from corehq.apps.accounting.tests import BaseAccountingTest
+from corehq.apps.accounting.tests.utils import DomainSubscriptionMixin
 from corehq.apps.domain.models import Domain
+from corehq.apps.domain.shortcuts import create_domain
+from corehq.apps.receiverwrapper.util import get_submit_url
+from corehq.apps.users.models import WebUser
+
+from corehq.elastic import get_es_new, send_to_elasticsearch
```
Comment (dimagimon) [inline corehq/apps/reports/tests/test_export_api.py]: F401 'send_to_elasticsearch' imported but unused

File: corehq/apps/reports/tests/test_export_api.py [hunk 10]
```diff
@@ -1,19 +1,32 @@
...
+from corehq.form_processor.utils import TestFormMetadata
+from corehq.pillows.xform import XFormPillow, get_sql_xform_to_elasticsearch_pillow
```
Comment (dimagimon) [inline corehq/apps/reports/tests/test_export_api.py]: F401 'get_sql_xform_to_elasticsearch_pillow' imported but unused

**Comment by dimagimon** [comment]
There are 11 errors:

* corehq/apps/reports/tests/test_esaccessors.py, line 665 - F841 local variable 'received_on_out' is assigned to but never used
* corehq/apps/reports/tests/test_esaccessors.py, line 699 - F841 local variable 'received_on_out' is assigned to but never used
* corehq/apps/reports/tests/test_esaccessors.py, line 678 - F841 local variable 'start' is assigned to but never used
* corehq/apps/reports/tests/test_esaccessors.py, line 679 - F841 local variable 'end' is assigned to but never used
* corehq/apps/reports/tests/test_esaccessors.py, line 692 - F841 local variable 'app2' is assigned to but never used
* corehq/apps/reports/tests/test_esaccessors.py, line 682 - F841 local variable 'received_on_out' is assigned to but never used
* corehq/apps/reports/tests/test_esaccessors.py, line 661 - F841 local variable 'start' is assigned to but never used
* corehq/apps/reports/tests/test_esaccessors.py, line 662 - F841 local variable 'end' is assigned to but never used
* corehq/apps/reports/tests/test_esaccessors.py, line 693 - F841 local variable 'xmlns2' is assigned to but never used
* corehq/apps/reports/tests/test_esaccessors.py, line 695 - F841 local variable 'start' is assigned to but never used
* corehq/apps/reports/tests/test_esaccessors.py, line 696 - F841 local variable 'end' is assigned to but never used

**Comment by emord** [comment]
Added tests for esaccessors. I wasn't able to get patching to work in descendants of  `TestCase`, so I just addes ES support for those

**Comment by benrudolph** [comment]
👍

**Comment by benrudolph** [comment]
cool all looks good on :chec

**Comment by emord** [comment]
```
======================================================================
FAIL: test_transfer_first_doc_order (corehq.apps.commtrack.tests.test_xml.CommTrackBalanceTransferTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/corehq/util/test_utils.py", line 230, in inner
    return helper(*args, **kwargs)
  File "/mnt/corehq/util/test_utils.py", line 203, in __call__
    call_with_settings(fn_with_pre_and_post, run_config.settings, args, kwargs)
  File "/mnt/corehq/util/test_utils.py", line 218, in call_with_settings
    fn(*args, **kwargs)
  File "/mnt/corehq/util/test_utils.py", line 199, in fn_with_pre_and_post
    self.fn(*args, **kwargs)
  File "/mnt/corehq/apps/commtrack/tests/test_xml.py", line 398, in test_transfer_first_doc_order
    self.check_product_stock(self.sp, product, final, 0)
  File "/mnt/corehq/apps/commtrack/tests/test_xml.py", line 227, in check_product_stock
    self.assertEqual(expected_soh, latest_trans.stock_on_hand)
AssertionError: Decimal('50.0') != 150
```

Not sure why this test is failing? doesn't seem related. it is skipped when I run it locally

**Comment by benrudolph** [comment]
i think it was broken before, rebasing with master fixed it for me

**Comment by emord** [comment]
:white_check_mark:

**Pull request merged**